webapi/src/test/scala/org/knora/webapi/store/triplestore/upgrade/plugins/UpgradePluginPR2255Spec.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.store.triplestore.upgrade.plugins
+
+import com.typesafe.scalalogging.LazyLogging
+
+import org.knora.webapi.messages.OntologyConstants
+import org.knora.webapi.messages.util.rdf._
+
+class UpgradePluginPR2255Spec extends UpgradePluginSpec with LazyLogging {
+  private val nodeFactory: RdfNodeFactory = RdfFeatureFactory.getRdfNodeFactory()
+
+  "Upgrade plugin PR2255" should {
+    "transform all project IRIs, that no old project IRI can be found in the data" in {
+      // Parse the input file.
+      val model: RdfModel = trigFileToModel("../test_data/upgrade/pr2255.trig")
+
+      val numberOfStatementsBeforeTransformation = model.size
+
+      // Use the plugin to transform the input.
+      val plugin = new UpgradePluginPR2255(log)
+      plugin.transform(model)
+
+      val numberOfStatementsAfterTransformation = model.size
+
+      // there should be the same number of statements before and after the transformation
+      assert(numberOfStatementsBeforeTransformation == numberOfStatementsAfterTransformation)
+
+      // Check projects IRIs was changed.
+      ProjectsIrisToChange.newToOldIrisMap.foreach { case (oldIri, _) =>
+        for {
+          statement <- model
+
+          // there shoudln't be any old project IRI in the data anymore, both as a subject and object
+          _ = assert(statement.subj.stringValue != oldIri)
+          _ = assert(statement.obj.stringValue != oldIri)
+        } yield ()
+      }
+    }
+
+    "not transform Standoff mappings, which IRIs contain project IRI" in {
```

Comment:
```suggestion
    "not transform Standoff mappings, whose IRIs contain a project IRI" in {
```

Reply:
I don't really understand why standoff mapping IRIs are not changed as well. Does it mean that they then contain a project IRI of the old form? This would not be conform to the documentation which states "However, the Iri of a mapping is predictable: **it consists of the project Iri** followed by /mappings/ and the knora-api:mappingHasName submitted in the JSON-LD" (see https://docs.dasch.swiss/2022.10.01/DSP-API/03-apis/api-v2/xml-to-standoff-mapping/). Then, this would not be true anymore.
Or maybe I don't understand this test ;)

Reply:
I think you understand the test, but long story short, other ITIs will be transformed in separate task.

Reply:
Standoff is actually special case, because it contains **exact** Project IRI and I'm still not sure if it shouldn't be transformed in this PR too, but this would need redesign of the plugin. The point is the `knora-iris.md` file was the base for me to work on this task and it was containing wrong data about Standoff IRIs. I fixed it, but because after creation of plugin. So I could update the docs you've linked above, but it will be again invalid the time we have all IRIs transformed... never ending story ;)

Reply:
I'm relatively confident that this doesn't break anything. The main reason that the mapping IRI is defined to contain exactly the project IRI plus a given suffix, is because in transformations you need to specify the mapping IRI, which hasn't been created at that point in time, so it needs to be predictable. But I don't think anything ever relies on the mapping IRI containing the correct project IRI within our system.

Reply:
OK, that's good. But then we should still update it in my opinion. If the design of mapping IRI is that it contains the project IRI, we should do it like that. Otherwise, I am sure that we (or whoever works with the data) will be very confused in the future :)
If this is done in the same plugin or in a next one, is of course another question.